pkg/archive/zip/zip_test.go

```diff
@@ -85,6 +86,13 @@ func TestZipFile(t *testing.T) {
 		}
 		if zf.Name == "link.txt" {
 			require.True(t, zf.FileInfo().Mode()&os.ModeSymlink != 0)
+			rc, err := zf.Open()
+			require.NoError(t, err)
+			var content bytes.Buffer
+			_, err = io.Copy(&content, rc)
+			require.NoError(t, err)
+			rc.Close()
+			require.Equal(t, content.String(), "regular.txt")
```

Comment:
I think here we can merely check that the link points to the expected file, no need to actually check file contents.

checking contents also is not guarantee that the link is correct, as it may point to another file with the same content

Reply:
Sorry, my variable name for the buffer might be confusing things. I am checking the destination of the link here. The destination path for the link is encoded as the contents of the link file in the archive. I'll update the variable name to help clarify this.

Reply:
oh, I totally misread it as well 😂